app/game-master/reducers/score.js

```diff
@@ -1,80 +1,34 @@
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc5', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc6', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc7', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc8', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc9', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc10', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc11', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc12', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc13', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc14', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc15', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc16', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc17', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc18', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc19', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc20', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc21', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc22', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc23', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc24', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc25', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc26', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc27', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc28', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc29', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc30', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc31', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc32', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc33', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc34', time: 25 },
-        { displayName: 'John Wick', participantId: 'qwe-asd-zxc35', time: 25 },
-    ],
-};
+import MockData from '../mock-data/score';
+import {
+    PARTICIPANT_JOINED,
+    PARTICIPANT_LEFT,
+    PARTICIPANT_SOLUTION
+} from '../actions/score';
+
+const defaultState = MockData;
 
 export default function participant(state = defaultState, action) {
-    switch(action.type) {
+    switch (action.type) {
+        case PARTICIPANT_JOINED:
+            return Object.assign({}, state, {
+                round: state.round.concat([action.payload]),
```

Comment:
It still can be immutable, but at least it may help with conditioning on render level (`.time || 0`, `.length || 0` etc)